worlds/inscryption/Options.py

```diff
@@ -0,0 +1,116 @@
+from dataclasses import dataclass
+
+from Options import Toggle, Choice, DeathLinkMixin, PerGameCommonOptions
+
+
+class Act1DeathLinkBehaviour(Choice):
+    """If DeathLink is enabled, determines what counts as a death in act 1. This affects deaths sent and received.
+    Sacrificed: Send a death when sacrificed by Leshy. Receiving a death will extinguish all candles.
+    Candle Extinguished: Send a death when a candle is extinguished. Receiving a death will extinguish a candle."""
+    auto_display_name = True
+    display_name = "Act 1 Death Link Behaviour"
+    option_sacrificed = 0
+    option_candle_extinguished = 1
+
+
+class Goal(Choice):
+    """Defines the goal to accomplish in order to complete the randomizer.
+    Full Story In Order: Complete each act in order. You can return to previously completed acts.
+    Full Story Any Order: Complete each act in any order. All acts are available from the start.
+    First Act: Complete Act 1 by finding the New Game button. Great for a smaller scale randomizer."""
+    auto_display_name = True
+    display_name = "Goal"
+    option_full_story_in_order = 0
+    option_full_story_any_order = 1
+    option_first_act = 2
+
+
+class RandomizeCodes(Toggle):
+    """Randomize codes and passwords in the game (clocks, safes, etc.)"""
+    display_name = "Randomize Codes"
+
+
+class RandomizeDeck(Choice):
+    """Randomize cards in your deck into new cards.
+    Disable: Disable the feature.
+    Every Encounter Within Same Type: Randomize cards within the same type every encounter (keep rarity/scrybe type).
+    Every Encounter Any Type: Randomize cards into any possible card every encounter.
+    Starting Only: Only randomize cards given at the beginning of runs and acts."""
+    auto_display_name = True
+    display_name = "Randomize Deck"
+    option_disable = 0
+    option_every_encounter_within_same_type = 1
+    option_every_encounter_any_type = 2
+    option_starting_only = 3
+
+
+class RandomizeSigils(Choice):
+    """Randomize sigils printed on the cards into new sigils every encounter.
+    Disable: Disable the feature.
+    Randomize Addons: Only randomize sigils added from sacrifices or other means.
+    Randomize All: Randomize all sigils."""
+    auto_display_name = True
+    display_name = "Randomize Abilities"
+    option_disable = 0
+    option_randomize_addons = 1
+    option_randomize_all = 2
+
+
+class OptionalDeathCard(Choice):
+    """Add a moment after death in act 1 where you can decide to create a death card or not.
+    Disable: Disable the feature.
+    Always on: The choice is always offered after losing all candles.
+    DeathLink Only: The choice is only offered after receiving a DeathLink event."""
```

Comment:
Out of curiosity, how does this work with DeathLinks only extinguishing _one_ candle? Does it have to extinguish the last one specifically?

Reply:
Yes, that'd be how it works.